File: app/src/main/java/com/cyl/musiclake/data/db/DBDaoLitepal.java
package com.cyl.musiclake.data.db;

import com.cyl.musiclake.MusicApp;
import com.cyl.musiclake.bean.Playlist;
import com.cyl.musiclake.bean.QueryInfo;
import com.cyl.musiclake.utils.ToastUtils;

import org.litepal.crud.DataSupport;

import java.util.List;

/**
 * Created by master on 2018/4/26.
 */

public class DBDaoLitepal {

    public void deletePlaylist(long pId) {
        int result = DataSupport.delete(Playlist.class, pId);
        if (result != 0) {
            ToastUtils.show(MusicApp.getAppContext(), "删除成功");
        }
    }

    public void addSearchInfo(String info) {
        QueryInfo queryInfo = new QueryInfo(info);
        queryInfo.save();
    }

    public List<QueryInfo> getSearchInfos() {
        return DataSupport.findAll(QueryInfo.class);
    }

}

File: app/src/main/java/com/cyl/musiclake/data/db/DBDaoLitepal.kt
package com.cyl.musiclake.data.db

import com.cyl.musiclake.MusicApp
import com.cyl.musiclake.bean.Playlist
import com.cyl.musiclake.bean.SearchHistoryBean
import com.cyl.musiclake.utils.ToastUtils

import org.litepal.crud.DataSupport

/**
 * 数据库操作类
 * Created by master on 2018/4/26.
 */

/**
 * 获取搜索历史
 */
fun getAllSearchInfo(): List<SearchHistoryBean> {
    return DataSupport.findAll(SearchHistoryBean::class.java)
}

/**
 * 删除歌单
 */
fun deletePlaylist(pId: Long) {
    val result = DataSupport.delete(Playlist::class.java, pId)
    if (result != 0) {
        ToastUtils.show(MusicApp.getAppContext(), "删除成功")
    }
}

/**
 * 增加搜索
 */
fun addSearchInfo(info: String) {
    val id = System.currentTimeMillis()
    val queryInfo = SearchHistoryBean(id, info)
    DataSupport.isExist(SearchHistoryBean::class.java, "title = $queryInfo")
}
